add image modes

foolab opened the pull request:
These series of commits add image-mode and supported-image-modes properties.
This enables 4 modes:
Normal image capture which is what we have been using so far
ZSL
HDR
ZSL and HDR at the same time

foolab committed:
droidcamsrc: Add image-mode and supported-image-modes properties

foolab committed:
droidcamsrc: get rid of fast-capture and fast-capture-supported properties

Use the new image-mode property instead. No guarantee that the code works yet.

foolab committed:
droidcamsrc: apply HDR quirk when needed

foolab committed:
droidcamsrc: implement missing bits for zsl_and_hdr image mode

saukko commented on gst/droidcamsrc/gstdroidcamsrc.c:
```diff
@@ -2207,3 +2226,53 @@ gst_droidcamsrc_find_picture_resolution (GstDroidCamSrc * src,
+
+
+static gboolean
+gst_droidcamsrc_is_zsl_and_hdr_supported (GstDroidCamSrc * src)
```
I feel that these should really be something like:
gst_droidcamsr_supported_modes() or something that would return array what is supported.

And similar thing for above gst_droidcamsr_enabled_modes() that would return what is enabled.

As if there will be more modes available in the future these will not really scale if you need to do separate functions for:
a_and_b
b_and_c
a_and_c
a_and_b_and_c
...

foolab replied:
Those are internal implementation details. Not part of the API offering at all and can be refactored anytime without affecting the API at all

saukko replied:
Not part of api offering, what does that mean? Nothing outside gst-droid uses it?

foolab replied:
The functions are implementation detail.
The 2 properties image-mode and supported-image-modes are the new additions to be used for interacting with upper layers.

foolab committed:
droidcamsrc: rework image-mode and supported-image-modes

image-mode is to use flags
supported-image-modes is to return a GArray of supported modes

This allows us to get rid of GST_DROIDCAMSRC_IMAGE_MODE_ZSL_AND_HDR
and allows adding future combinations

abranson commented:
LGTM!
